File: parking-web-common/src/main/java/com/chatchatabc/parking/web/common/application/rest/service/JwtService.java
package com.chatchatabc.parking.web.common.application.rest.service;

import com.auth0.jwt.interfaces.Payload;

import java.util.List;

public interface JwtService {

    /**
     * Generate a JWT token for the given user id
     *
     * @param userId   the user id
     * @param username the username
     * @param roles    the roles
     * @return the JWT token
     */
    String generateToken(String userId, String username, List<String> roles);

    /**
     * Validate the given token and return the jwt payload
     *
     * @param token the token
     * @return the jwt payload
     */
    Payload validateTokenAndGetPayload(String token);
}

File: backend/src/main/kotlin/com/chatchatabc/parking/application/rest/service/JwtService.kt
package com.chatchatabc.parking.application.rest.service

import com.chatchatabc.parking.domain.model.User
import org.springframework.stereotype.Service

@Service
interface JwtService {
    /**
     * Generate Token
     */
    fun generateToken(user: User): String

    /**
     * Validate Token and Get User
     */
    fun validateTokenAndGetUser(token: String): User
}